zerver/tests/test_push_registration.py

```diff
@@ -190,3 +197,192 @@ def test_register_push_device_error(self) -> None:
             invalid_token_payload,
         )
         self.assert_json_error(result, "Invalid encrypted_push_registration")
+
+
+class RegisterPushDeviceToServer(ZulipTestCase):
+    @staticmethod
+    def get_register_push_device_payload() -> dict[str, str | int]:
+        payload: dict[str, str | int] = {
+            "token_kind": PushDevice.TokenKind.FCM,
+            "push_account_id": 2408,
+            "push_public_key": "dummy-push-public-key",
+            "bouncer_public_key": "bouncer-public-key",
+            "encrypted_push_registration": "encrypted-push-registration",
+        }
+        return payload
+
+    @mock.patch(
+        "zerver.lib.push_registration.do_register_remote_push_device",
+        return_value=3,
+    )
+    def test_register_push_device_success(self, unused_mock: mock.Mock) -> None:
+        self.login("hamlet")
+
+        push_devices_count = PushDevice.objects.count()
+        self.assertEqual(push_devices_count, 0)
+
+        payload = self.get_register_push_device_payload()
+
+        # Verify the created PushDevice row while the registration
+        # to the bouncer is in progress (worker hasn't consumed yet).
+        with mock.patch("zerver.views.push_notifications.queue_event_on_commit"):
+            result = self.client_post("/json/mobile_push/register", payload)
```

Comment:
This isn't testing that something in fact got enqueued; that seems like a thing we should do?